src/main/java/teammates/ui/webapi/action/GetInstructorsAction.java

```diff
@@ -76,7 +77,18 @@ public ActionResult execute() {
             });
         } else if (intentStr.equals(Intent.FULL_DETAIL.toString())) {
             // get all instructors of a course without information hiding
-            data = new InstructorsData(instructorsOfCourse);
+            // adds googleId if caller is course co-owner or admin
+            if (userInfo.isAdmin || logic.getInstructorForGoogleId(courseId, userInfo.getId()).getRole().equals(
+                    Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER)) {
```

Comment:
After much debugging I've isolated the problem with the e2e tests to the addition of this line. Whenever `verifyPresentInDatastore` is called in a test case for instructors, the `BackDoor` will attempt to execute a `GET` request on the `/instructors` endpoint. This will now however always throw a NPE because `userInfo.getId()` is null. Any advice to help resolve this would be appreciated @wkurniawan07 @xpdavid @jtankw3

Reply:
I was hoping against that because the `getMinAuthLevel` is already defined as `LOGGED_IN`. There appears to be another issue as well with `UpdateInstructorAction`. I'm working with @jtankw3 to help resolve it

Reply:
By design, the `execute()` method should not care about access control (this is the complete seperation of access control we want to achieve in V7) so we are not breaking any promise.